### PR TITLE
Scattergl transform fixes

### PR DESCRIPTION
**Pull request by etpinard**
All transforms are broken on scattergl traces. Yikes!

Luckily it wasn't too hard to fix them. See https://github.com/plotly/plotly.js/commit/c52fbe75379073a9bc69292f18088e73e4e5f977 for more details. In the meantime, I also discovered two other bugs that made the `transforms` and `point-selection` mocks look (very) bad) using scattergl traces instead on scatter traces. This PR adds scattergl version of these mocks :lock: ing down the behavior.

cc @alexcjohnson (I'll add a jasmine test tonight or tomorrow morning before 1.38.1)

### CODE REVIEW

File: src/traces/scattergl/convert.js [hunk 1]
```diff
@@ -192,10 +192,10 @@ function convertMarkerStyle(trace) {
         // See  https://github.com/plotly/plotly.js/pull/1781#discussion_r121820798
         if(multiLineWidth) {
             for(i = 0; i < count; i++) {
-                borderSizes[i] = markerSizeFunc(optsIn.line.width[i]);
+                borderSizes[i] = optsIn.line.width[i] / 2;
             }
         } else {
-            s = markerSizeFunc(optsIn.line.width);
+            s = optsIn.line.width / 2;
```
Comment (etpinard) [inline src/traces/scattergl/convert.js]: Before this commit, `gl2d_transforms` looked like:

image

File: src/traces/scattergl/convert.js [hunk 2]
```diff
@@ -219,7 +219,7 @@ function convertMarkerSelection(trace, target) {
     if(target.marker && target.marker.symbol) {
         optsOut = convertMarkerStyle(Lib.extendFlat({}, optsIn, target.marker));
     } else if(target.marker) {
-        if(target.marker.size) optsOut.sizes = target.marker.size;
+        if(target.marker.size) optsOut.sizes = target.marker.size / 2;
```
Comment (etpinard) [inline src/traces/scattergl/convert.js]: Before this commit, `gl2d_point-selection` looked like:

image

**Comment by etpinard** [comment]
Marking as `reviewable`.

**Comment by alexcjohnson** [comment]
Very nice! Great catches on the size and linewidth issues! 💃

**Pull request merged**